@xen-orchestra/xapi/host.mjs

```diff
@@ -120,6 +180,39 @@ class Host {
     await this.callAsync('host.reboot', ref)
     await waitAgentRestart(this, ref, agentStartTime)
   }
+
+  async getIpmiSensors(ref) {
+    const productName = (await this.getField('host', ref, 'bios_strings'))['system-product-name']?.toLowerCase()
+    if (IPMI_SENSOR_REGEX_BY_DATA_TYPE_BY_SUPPORTED_PRODUCT_NAME[productName] === undefined) {
+      return {}
+    }
+    const callSensorPlugin = fn => this.call(IPMI_CACHE, 'host.call_plugin', ref, '2crsi-sensors.py', fn, {})
```

Comment:
IMHO, the cache should be passed by the caller (in a `{ cache } = {}` param), like we did in `xen-api`, otherwise we'll have the same problems when we want to not use a cache or refresh it.

Reply:
Maybe I should handle an `undefined` cache in `xen-api` to simplify the calling code.

Reply:
`[...] passed by the caller`.
Who do you think is the `caller`?
The `getIpmiSensors` method or the `callSensorPlugin` wrapper?

Reply:
The caller of `getIpmiSensors`.